packages/data-service/src/data-service.ts

```diff
@@ -1513,6 +1534,51 @@ class DataServiceImpl extends WithLogContext implements DataService {
     return await coll.dropIndex(name);
   }
 
+  @op(mongoLogId(1_001_000_237))
+  async isListSearchIndexesSupported(ns: string): Promise<boolean> {
+    try {
+      await this.getSearchIndexes(ns);
+    } catch (err) {
+      return false;
+    }
+    return true;
+  }
+
+  @op(mongoLogId(1_001_000_238))
+  async getSearchIndexes(ns: string): Promise<SearchIndex[]> {
+    const coll = this._collection(ns, 'CRUD');
+    const cursor = coll.listSearchIndexes();
+    const indexes = await cursor.toArray();
+    void cursor.close();
+    return indexes as SearchIndex[];
```

Comment:
Why do we need to assert here? Should we raise with the driver that the types are wrong on their side?

Reply:
We could? The type is just Document. Same happens for regular indexes: https://github.com/mongodb-js/compass/blob/3e592c820facff86318ec4a9aaf3a0daa565787b/packages/data-service/src/data-service.ts#L1808

Reply:
It's not `Document`, `listSearchIndexes` returns `ListSearchIndexesCursor` that has only `{ name: string; }` property, there must be a reason other properties are not returned from driver types?

Reply:
To be clear I think it's totally fine to keep the assertion in our code right now, just seems like driver wants to return a specific type here, maybe it's worth checking with them why and change it on their side if that's a type issue in the driver

Reply:
Aah I see now. We'll see what they say and then we can update this. The info comes from an aggregation so I guess _something_ either in the driver or on our side will ultimately have to assert.